cts/cts-cli.in

```diff
@@ -1886,6 +1898,20 @@ function test_feature_set() {
     unset CIB_shadow_dir
 }
 
+function test_tool_cmdline() {
+    desc="crm_resource run with extra arguments"
+    cmd="crm_resource foo bar"
+    test_assert $CRM_EX_USAGE 0
+
+    desc="crm_resource given both -r and resource config"
+    cmd="crm_resource -r xyz --class ocf --provider pacemaker --agent Dummy"
+    test_assert $CRM_EX_USAGE 0
+
+    desc="crm_resource given resource config with invalid action"
+    cmd="crm_resource --class ocf --provider pacemaker --agent Dummy -D"
+    test_assert $CRM_EX_USAGE 0
+}
+
```

Comment:
Why separate this from tools?

Reply:
No real compelling reason.  I was just thinking about keeping tools from growing without bounds forever.  But maybe it doesn't matter.

Reply:
The main effect is allowing someone to run a subset of the tests with `cts-cli -t`. Aside from initial development, I don't see why someone would want to run just these tests or just the tools tests without these, so I'm leaning to not separating it.

The current division of tests is kind of questionable too, but not worth spending much time on. I would think most of the time we just want to run all the tests, and if not, it's likely because we want to test only the code affected by a change we're working on. From that view, the dates, crm_mon, acls, and rules divisions make sense. The rest probably don't, but I could see maybe separating everything that works on a test CIB from everything that doesn't.